### ISSUE TITLE
MenuSheetView is not working in GRID mode

### ISSUE THREAD

**Issue by demogorgorn** [Flipboard/bottomsheet#19]
Hello! MenuSheetView isn't showing the icons in grid mode. My code

```java

MenuSheetView menuSheetView =
                new MenuSheetView(getActivity(), MenuSheetView.MenuType.LIST, "Create...", new MenuSheetView.OnMenuItemClickListener() {
                    @Override
                    public boolean onMenuItemClick(MenuItem item) {
                        Toast.makeText(getActivity(), item.getTitle(), Toast.LENGTH_SHORT).show();
                        if (bottomSheetLayout.isSheetShowing()) {
                            bottomSheetLayout.dismissSheet();
                        }
                        return true;
                    }
                });
        menuSheetView.inflateMenu(R.menu.menu_sheet_add_plan);
        bottomSheetLayout.showWithSheetView(menuSheetView);

```

```xml

<?xml version="1.0" encoding="utf-8"?>
<menu xmlns:android="http://schemas.android.com/apk/res/android">
<item
    android:id="@+id/plan_type_payment"
    android:icon="@drawable/creditcard"
    android:title="Payment"/>

<item
    android:id="@+id/menu_subheader"
    android:title=""
    >
    <menu>
        <item
            android:id="@+id/plan_type_plane"
            android:icon="@drawable/plane"
            android:title="Plane"/>
        <item
            android:id="@+id/plan_type_train"
            android:icon="@drawable/train"
            android:title="Train"/>
        <item
            android:id="@+id/plan_type_car"
            android:icon="@drawable/car"
            android:title="Car"/>
        <item
            android:id="@+id/plan_type_bus"
            android:icon="@drawable/schooolbus"
            android:title="Bus"/>
        <item
            android:id="@+id/plan_type_taxi"
            android:icon="@drawable/taxi"
            android:title="Taxi"/>
        <item
            android:id="@+id/plan_type_cruise"
            android:icon="@drawable/cruise"
            android:title="Cruise"/>
        <item
            android:id="@+id/plan_type_sailboat"
            android:icon="@drawable/sailboat"
            android:title="Sailboat"/>
        <item
            android:id="@+id/plan_type_bike"
            android:icon="@drawable/bike"
            android:title="Bike"/>
    </menu>
</item>

</menu>

```

**Comment by demogorgorn** [comment]
I've found where is the problem. I'm trying to use MenuSheetView in my fragment. And the onAttachedToWindow function is not called and the adapter is not initialized! Can you please help me! Thanks!

**Comment by demogorgorn** [comment]
GRID mode mode not working also in the sample app.

**Comment by hzsweers** [comment]
GRID mode works fine for me, what's not working for you?

Also, that menu xml works fine for me.

![victara_tmolpe23 32-21 3hsweers07132015134805](https://cloud.githubusercontent.com/assets/1361086/8660484/0e8a4808-2966-11e5-935b-3c69c73b1464.png)

**Comment by demogorgorn** [comment]
Android Studio 
android {
    compileSdkVersion 22
    buildToolsVersion "22.0.1"

    defaultConfig {
        minSdkVersion 15
        targetSdkVersion 22
        versionCode 1
        versionName "1.0"
    }

Your Sample and my app are not showing icons in grid. My app is using fragments. So when I switch to other fragment and than turn back to previous where is opened! grid - all icons are visible.

Temporarily switched to https://github.com/soarcn/BottomSheet

**Comment by hzsweers** [comment]
It shouldn't matter what fragment you're using, bottomsheet is attached to the activity context. That library's implementation is via dialog, which doesn't help debug this either.

What kind of device are you running it on? Android version? Can you breakpoint the sheet and see if the 
`SheetMenuItem#getMenuItem()` at any of those positions returns something for getIcon()? It's a really [simple setup](https://github.com/Flipboard/bottomsheet/blob/master/bottomsheet-commons/src/main/java/com/flipboard/bottomsheet/commons/MenuSheetView.java#L349) with a standard viewholder pattern, so I don't see why this behaves differently than any other gridview.

**Comment by demogorgorn** [comment]
I've already noticed that MenuSheetView is quite simple and understandable. That's why I don't understand where is the problem. My device is Sony Ericsson Xperia Neo V with android 4.0.4.

I'll try later to follow your suggestions on breakpoint. Strange is that if I switch to other fragment and than turn back to previous where is opened! grid - all icons are visible. 

Thanks for help

**Comment by st028** [comment]
@demogorgorn we found the source of this issue and fixed it in #41

**Comment by demogorgorn** [comment]
Aha! Thank you very much!
